meson.build

```
@@ -186,6 +186,28 @@ if not get_option('snpe-support').disabled()
   endif
 endif
 
+# tensorrt
+tensorrt_dep = dependency('', required: false)
```

Is this effective line?

---

I think a discussion made at https://github.com/nnstreamer/nntrainer/pull/150#discussion_r439172394 might help this line.

---

@zhoonit I could not still get it. 
```suggestion
tensorrt_dep = disabler()
```

Do you mean this?

---

I am not sure if a disabler object acts like a dependency object as the reference states that the disabler object has only one method: `found()`. If it is fine to pass a disabler object to a `build_target()`, it is equivalent.

---

@zhoonit If tensorrt_dep.fount() is false, why do we provide this tensorrt_dep to build_target()?

---

I don't know the exact use case only seeing this line. So I am not sure if `disabler()` can substitute `dependency('', required:false)`

What I know is `found()` is required in this file which makes this line effective.

Although I can think of a usecase for passing not_found to the build_target

```meson
A = dependency('foo', required: false)
B = executable('bar', dependencies: A, fallback: [......])
```
or

```meson
nnfw_dep = dependency('', required: false)
if not get_option('nnfw-runtime-support').disabled()
  nnfw_dep = dependency('nnfw', required: false)
  if not nnfw_dep.found()
    nnfw_dep = cc.find_library('nnfw-dev', required: get_option('nnfw-runtime-support'))
  endif
endif
```
and adding `nnfw_dep` to a `build_target()` will safely throw when needed

---

@zhoonit
I hardly think so.
https://mesonbuild.com/Disabler.html

https://git.sailfishos.org/mirror/rygel/blob/134eb29fa1994ca2e97617579a28de0fb69d39bf/meson.build

disabler() is introduced for reducing such fragmented if-else-endif.

---

I think `disabler()` looks good. BTW, I have modified this PR to use `tensorrt_deps` not `tensorrt_dep`.

---

Yes I think `disabler()` can replace the given line :)